src/main/kotlin/dev/minn/jda/ktx/JDABuilder.kt

```diff
@@ -0,0 +1,399 @@
+        builder.disableCache(cacheFlag, *cacheFlags)
+    }
+
+    inline fun disableCache(cacheFlags: Set<CacheFlag>) {
+        builder.disableCache(cacheFlags)
+    }
+
+    var memberCachePolicy: MemberCachePolicy? = null
+        set(value) {
+            builder.setMemberCachePolicy(value)
+            field = value
+        }
+
+    var contextMap: ConcurrentMap<String, String>? = null
+        set(value) {
+            builder.setContextMap(value)
+            field = value
+        }
+
+    var contextEnabled: Boolean = true
+        set(value) {
+            builder.setContextEnabled(value)
+            field = value
+        }
+
+    var compression: Compression? = null
+        set(value) {
+            if (value != null) {
+                builder.setCompression(value)
+            }
+            field = value
+        }
+
+    var requestTimeoutRetry: Boolean = true
+        set(value) {
+            builder.setRequestTimeoutRetry(value)
+            field = value
+        }
+
+    var token: String? = null
+        set(value) {
+            builder.setToken(value)
+            field = value
+        }
+
+    var httpClientBuilder: OkHttpClient.Builder? = null
+        set(value) {
+            builder.setHttpClientBuilder(value)
+            field = value
+        }
+
+    var httpClient: OkHttpClient? = null
+        set(value) {
+            builder.setHttpClient(value)
+            field = value
+        }
+
+    var webSocketFactory: WebSocketFactory? = null
+        set(value) {
+            builder.setWebsocketFactory(value)
+            field = value
+        }
+
+    var rateLimitPool: ScheduledExecutorService? = null
+        set(value) {
+            builder.setRateLimitPool(value, true)
+            field = value
+        }
+
+    var gatewayPool: ScheduledExecutorService? = null
+        set(value) {
+            builder.setGatewayPool(value, true)
+            field = value
+        }
+
+    var callbackPool: ExecutorService? = null
+        set(value) {
+            builder.setCallbackPool(value, true)
+            field = value
+        }
+
+    var eventPool: ExecutorService? = null
+        set(value) {
+            builder.setEventPool(value, true)
+            field = value
+        }
+
+    var audioPool: ScheduledExecutorService? = null
+        set(value) {
+            builder.setAudioPool(value, true)
+            field = value
+        }
+
+    var bulkDeleteSplittingEnabled: Boolean = true
+        set(value) {
+            builder.setBulkDeleteSplittingEnabled(value)
+            field = value
+        }
+
+    var enableShutdownHook: Boolean = true
+        set(value) {
+            builder.setEnableShutdownHook(value)
+            field = value
+        }
+
+    var autoReconnect: Boolean = true
+        set(value) {
+            builder.setAutoReconnect(value)
+            field = value
+        }
+
+    var eventManager: IEventManager? = null
```

Comment:
The thing is when this event manager is changed and injectKtx is true that will cause the manager to be ignored, maybe there is a better way to handle that?

Reply:
Hmmm, fair point.

Here's perhaps another idea:
rewrite the injected manager slightly so that it takes an event manager in the constructor. Then, it will delegate the event registration to the new even manager and if it fails, it sends registers it on its own.